var/spack/repos/builtin/packages/enzo/package.py

```diff
@@ -53,16 +62,29 @@ def edit(self, spec, prefix):
             filter_file('^LOCAL_HYPRE_INSTALL.*',
                         'LOCAL_HYPRE_INSTALL =',
                         'Make.mach.spack')
+            if spec.satisfies('%gcc@10.2.0:'):
+                filter_file(r'^MACH_FFLAGS\s*=\s*',
+                            'MACH_FFLAGS   = -fallow-argument-mismatch ',
+                            'Make.mach.spack')
+                filter_file(r'^MACH_F90FLAGS\s*=\s*',
+                            'MACH_F90FLAGS = -fallow-argument-mismatch ',
+                            'Make.mach.spack')
```

Comment:
Why is this needed? We already set the flag in `flag_handler` above.

Reply:
Well, I really don't know why, but in my setup with gcc 10.2 and without explicit modifications to MACH_F90FLAGS or MACH_FFLAGS it does not work. Got "`Error: Type mismatch in argument 'ispec' at (1); passed INTEGER(4) to INTEGER(8)`"

May be `flag_handler` is not good enough for custom enzo makefile code which seems to use several dialects of fortran at a same time.

Reply:
That's weird. I just removed this part and it still compiles for me (with GCC 11). I think it would be good to find out what's going on there.

Reply:
Weird, flag_handler is working for me too now